### ISSUE TITLE
Switching Light and Dark Mode - Suggested Idea

### ISSUE THREAD

**Issue by kaleb-keny** [spyder-ide/spyder#12624]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
Basically my windows switches between dark and light themes depending on the time of the day. I believe MACs have this options too. Would it be possible that upon launching Spyder, the windows set theme would be applied. Meaning I pre-set a dark and light theme in Spyder. When spyder launches, if the user had selected this auto-theme mode, then it'll fetch that pre-set light theme if the windows theme is light or set it to dark if the theme is dark.

**Comment by dalthviz** [comment]
Hi @kaleb-keny thanks for the suggestion! @spyder-ide/core-developers what do you guys thing about this?

**Comment by bcolsen** [comment]
Duplicate for mac https://github.com/spyder-ide/spyder/issues/10026

The interface theme is set to `automatic` by default I assumed we were already doing this. If not what does `automatic` mean?

There is also the issue of selecting the right syntax highlighter. When automatically switching. Ww could have two syntax highlighter pulldowns. One to use for the light themes and one for dark.
Talked about here: https://github.com/spyder-ide/spyder/issues/8317

**Comment by dalthviz** [comment]
Thanks @bcolsen for the reference! 

Also, regarding the current `automatic` option, as far as I remember it means that the `Interface theme` (`Dark` or `Light`) will be consistent with the selected `Syntax highlighting theme`. So, for example, if you're currently using the `Spyder Dark` syntax highlighting but then you change to `Spyder`, the Interface theme will change from dark to light and a restart will be requested.
